public/canvas/canvas-customization.js

```diff
@@ -0,0 +1,864 @@
+// jscs:disable maximumNumberOfLines
+/* jshint camelcase: false */
+
+(function(window, document, $) {
+  'use strict';
+
+  /* UTIL */
+
+  /**
+   * Perform a CalCentral API Get request
+   *
+   * @param  {String}     url                 The relative URL of the API request that should be made
+   * @param  {Function}   callback            Standard callback function
+   * @param  {Object}     callback.response   The API request response
+   */
+  var apiRequest = function(url, callback) {
+    $.ajax({
+      'dataType': 'json',
+      'url': window.CALCENTRAL + url,
+      'success': callback
+    });
+  };
+
+  /**
+   * Get the id for a custom LTI tool that user has access to
+   *
+   * @param  {String}     toolType            The classification of the LTI tool. This will be the top level property in the external tools hash
+   * @param  {String}     toolName            The name of the LTI tool for which the id should be retrieved
+   * @param  {Function}   callback            Standard callback function
+   * @param  {String}     callback.id         The id of the requested custom LTI tool
+   */
+  var getExternalToolId = function(toolType, toolName, callback) {
+    apiRequest('/api/academics/canvas/external_tools.json', function(externalToolsHash) {
```

Comment:
@mlincoln - this is a Ripley API reference and it should be changed to `/api/canvas/external_tools`